examples/examples/jsonrpsee_server_low_level_api.rs

```diff
@@ -0,0 +1,286 @@
+		let mut i = 0;
+		let handle = run_server();
+
+		let client = WsClientBuilder::default().build("ws://127.0.0.1:9944").await.unwrap();
+		while client.is_connected() {
+			let rp: Result<String, _> = client.request("say_hello", rpc_params!()).await;
+			if rp.is_ok() {
+				i += 1;
+			}
+		}
+
+		// After the server has blacklisted the IP address, the connection is denied.
+		assert!(WsClientBuilder::default().build("ws://127.0.0.1:9944").await.is_err());
+		tracing::info!("WS client made {i} successful calls before getting blacklisted");
+
+		handle.stop().unwrap();
+		handle.stopped().await;
+	}
+
+	// Make a bunch of HTTP calls to be blacklisted by server.
+	{
+		let mut i = 0;
+		let handle = run_server();
+
+		let client = HttpClientBuilder::default().build("http://127.0.0.1:9944").unwrap();
+		while let Ok(_) = client.request::<String, _>("say_hello", rpc_params![]).await {
+			i += 1;
+		}
+		tracing::info!("HTTP client made {i} successful calls before getting blacklisted");
+
+		handle.stop().unwrap();
+		handle.stopped().await;
+	}
+
+	Ok(())
+}
+
+fn to_jsonrpsee_params(
+	methods: impl Into<Methods>,
+	stop_handle: StopHandle,
+	conn_id: u32,
+	conn_permit: OwnedSemaphorePermit,
+) -> Params {
+	let cfg = Settings {
+		max_connections: 100,
+		max_request_body_size: 1024 * 1024,
+		max_response_body_size: 1024 * 1024,
+		max_subscriptions_per_connection: 128,
+		message_buffer_capacity: 1024,
+		enable_http: true,
+		enable_ws: true,
+		tokio_runtime: None,
+		ping_config: PingConfig::WithoutInactivityCheck(Duration::from_secs(30)),
+		batch_requests_config: jsonrpsee::server::BatchRequestConfig::Disabled,
+		id_provider: Arc::new(RandomIntegerIdProvider),
+	};
+
+	Params { methods: methods.into(), stop_handle, conn_id, conn_permit: Arc::new(conn_permit), cfg }
+}
+
+fn run_server() -> ServerHandle {
+	use hyper::service::{make_service_fn, service_fn};
+
+	// Construct our SocketAddr to listen on...
+	let addr = SocketAddr::from(([127, 0, 0, 1], 9944));
+
+	// Maybe we want to be able to stop our server but not added here.
+	let (tx, rx) = tokio::sync::watch::channel(());
+
+	let stop_handle = StopHandle::new(rx);
+	let server_handle = ServerHandle::new(tx);
+	let methods = ().into_rpc();
+	let conn_guard = ConnectionGuard::new(100);
+	let conn_id = Arc::new(AtomicU32::new(0));
+	let http_rate_limit = Arc::new(AsyncMutex::new(0));
+
+	// Blacklisted peers
+	let blacklisted_peers = Arc::new(Mutex::new(HashSet::new()));
+	let stop_handle2 = stop_handle.clone();
+
+	// And a MakeService to handle each connection...
+	let make_service = make_service_fn(move |conn: &AddrStream| {
+		// Connection state.
+		let conn_id = conn_id.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+		let remote_addr = conn.remote_addr();
+		let stop_handle = stop_handle2.clone();
+		let conn_guard = conn_guard.clone();
+		let methods = methods.clone();
+		let blacklisted_peers = blacklisted_peers.clone();
+		let http_rate_limit = http_rate_limit.clone();
```

Comment:
Maybe worth a note that the HTTP rate limit is global (because new connection every time) but the WS rate limit is per connection?